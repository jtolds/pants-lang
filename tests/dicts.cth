Dictionary := { construct { |obj|
  obj.~index = { |key|
    print "index" key
    (+ "index value " key)
  }
  obj.~update = { |key, value|
    print "update" key value
    (+ "update value " key " " value)
  }
}}

x := {"hey": "there", "dictionaries": "are cool"}

print x["groovy"]
print (x["test1"] = "test2")
