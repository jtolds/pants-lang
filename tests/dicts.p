x = { 3: 1, "hey": "there"}

print x[3] x[2]

x[3] = 4
x[2] = 5

print x[3] x[2] x["hey"] x["no"]

x[2] = 6

print x[2]
print x["hey"]

print x.has_key?("hey")
print x.has_key?("there")

x.each {|k, v| print "each:" k v }
print x.size()

y = x.copy.

y[10] = "new"
y[3] = 11

print "=="
x.each {|k, v| print "each:" k v }
print x.size()
print "=="
y.each {|k, v| print "each:" k v }
print y.size()

print.

print y.delete(3)
y.each {|k, v| print "each:" k v }
print y.size()
