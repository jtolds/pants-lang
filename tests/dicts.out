update hey there
update dictionaries are cool
index groovy
index value groovy
update test1 test2
test2
